src/main/scala/is/hail/variant/VariantDataset.scala

```diff
@@ -309,7 +329,54 @@ g = let newgt = ${ filterGT("gtIndex(oldToNew[gtj(g.gt)], oldToNew[gtk(g.gt)])")
  in Genotype(newV, Call(newgt), newad, newdp, newgq, newpl)
         """
       }
+    } else {
+      if (!vsm.genotypeSignature.isOfType(Genotype.htsGenotypeType))
+        fatal(s"filter_alleles: genotype_schema must be TGenotype or the HTS genotype schema, found: ${ vsm.genotypeSignature }")
+
+      if (subset) {
+        """
+g = let newpl = if (isDefined(g.PL))
+        let unnorm = range(newV.nGenotypes).map(newi =>
+            let oldi = gtIndex(newToOld[gtj(newi)], newToOld[gtk(newi)])
+             in g.PL[oldi]) and
```

Comment:
These expressions look right. The indentation is strange in places like here.

Reply:
I'm going to leave it.  We don't really have an indentation convention and it is all going to become python soon anyway.